### ISSUE TITLE
Add stylistic sets or new ligatures for = and ==

### ISSUE THREAD

**Issue by BenjaminGalliot** [cormullion/juliamono#111]
Hello, this request is closely related to these topics:
https://discourse.julialang.org/t/aliases-for-and/60004
https://discourse.julialang.org/t/my-brain-is-hard-wired-against-help-me-julia/63532
https://discourse.julialang.org/t/the-hegemony-of-must-end-welcome-the-upstart/63605

Some people prefer to have `=` as the equality operator rather than `==`, but it thus clashes with the assignment operator, which could be `←` rather than `=`. 

A solution, perhaps more elegant for some, at least without external incompatibility, would be to have local font options, such as special ligatures (to replace `==` with `=`) and stylistic sets (to turn `=` into `←` or even `:=`) to visually replace these operators with single-character size substitutes.

Would it be possible to dig in this way?

**Comment by cormullion** [comment]
tl;dr: It's technically possible, but I don't think I'll be implementing it any time soon.

1: I think the task of replacing "==" with an alternative pair of glyphs could be added to a Stylistic Set without too many problems. It would be opt-in, and supported if your editing environment can handle Stylistic Sets. Distinguishing between "==" and "===" might be a problem, but I think it's possible.

2: It's also possible for a font to replace `=` with some other glyph shape. The problem is that the simple equals sign typically works very hard in computer code, and it doesn't always mean assignment. For example, in Julia:

| Syntax        | Assignment?    |
| ------------- |:--------------:|
| `x = 3 `                  | ✓ |
| `x=3 `                    | ✓ |
| `x<<=3`                   | ✓ |
| `x===3`                   | ✗ |
| `x == 3`                  | ✗ |
| `x<=3 `                   | ✗ |
| `3>=x`                  | ✗ |
| `push!(a, '=')`           | ✗ |
| `#= comment`              | ✗ |
| `comment =#`              | ✗ |
| `match(r"q(?=u)", "")`    | ✗ |
| `x => 3`                  | ✗ |
| `"the = glyph..."`        | ✗ |
| `x += 3 `                 | ✓ (but to `x`, not to `+` ?)|
| `x .+= 3`                 | ✓ (but to `x`, not to `.+` ?)|

and I wondered about the following uses too:

```julia
julia> pi
π = 3.1415926535897  # what is the value of π
```

```julia
julia> match(r"^\s*(?:#\s*(.*?)\s*$|$)", "# a comment ")
RegexMatch("# a comment ", 1="a comment") # not an assignment to `1`?
```

```julia
for i = 1:2 # `=` is also interpreted as `in`?
```

And that's just Julia. Looking at the stars for this repo, it's likely that not all users of this font use Julia. So there are probably a number of additional uses to consider. For example, there are languages in which "<=" is an assignment... 😅

It would be certainly be possible to define a set of regular expressions to find all these. Not a job I'd feel happy tackling. But that wouldn't help much anyway, because OpenType fonts don't use regular expressions. The pattern matching tools available can do quite a lot, but handling all these wouldn't be trivial. And ultimately, I don't think the benefits of this feature are worth the trouble - there are loads of people who hate all "ligatures"!

**Comment by BenjaminGalliot** [comment]
Nice catch for point 2. I hadn't thought of many of these cases since I always use Unicode characters (`≤`, `≥`, `∈`, `∉`, etc.) when possible.

Anyway, it was mostly to see if there were any possibilities on the font path...

**Comment by cormullion** [comment]
hopefully this will make it into the next (0.41) Release.

![ligatures](https://user-images.githubusercontent.com/52289/123786759-f8457c80-d8d1-11eb-9119-65edb3a9fc44.gif)

**Comment by cormullion** [comment]
The ligature has to occupy the same amount of space as the characters it replaces, though… the font doesn’t change the text.

**Comment by BenjaminGalliot** [comment]
It's true, I was thinking of tricks related to modifying characters (diacritics...) and other zero-width characters, but this is directly related to the characters themselves and the font won't change anything...
